msteams-platform/bots/how-to/conversations/conversation-messages.md

```diff
@@ -137,6 +137,119 @@ async def on_message_activity(self, turn_context: TurnContext):
+
+#### RSC permission for read receipts 
+
+Add the [RSC](~/graph-api/rsc/resource-specific-consent.md#rsc-permissions-for-a-chat-or-meeting) `ChatMessageReadReceipt.Read.Chat` permission in the [app manifest](~/resources/schema/manifest-schema.md), as follows: 
+
+# [App manifest v1.11 or earlier](#tab/app-manifest-v111-or-earlier)
+
+```json
+
+ “webApplicationInfo”: {
+
+     "id": "123456c8-67d2-4f54-b74e-408b195c4cbc",
+     "resource": "https: //AnyString",
+     "applicationPermissions": [
+         "ChatMessageReadReceipt.Read.Chat"
+     ]
+ },
+
+```
+
+# [App manifest v1.12 or later](#tab/app-manifest-v112-or-later)
+
+```json
+
+"webApplicationInfo": {
+
+    "id": "38f0ca43-1c38-4c39-8097e-47f62c686500",
+    "resource": ""
+},
+"authorization": {
+    "permissions": {
+        "orgwide": [],
+        "resourceSpecific": [
+        {
+            "name": "ChatMessageReadReceipt.Read.Chat",
+            "type": "Application"
+        }
+        ]
+    }
+}
+
+```
+---
+
+You can also add RSC permissions through Graph API. For more information, see [`consentedPermissionSet`](/graph/api/userteamwork-teamsappinstallation-upgrade#http-request).
+
+#### Override `OnTeamsReadReceiptAsync`
+
+The `IsMessageRead` helper method is useful to determine if the message has been read. If the `compareMessageId` is less than or equal to the `LastReadMessageId`, then the message has been read. Override the `OnTeamsReadReceiptAsync` method to receive read receipts with [`IsMessageRead`](/dotnet/api/microsoft.bot.schema.teams.readreceiptinfo.ismessageread#microsoft-bot-schema-teams-readreceiptinfo-ismessageread(system-string)) helper method:
+
+```javascript
+
+protected override async Task OnTeamsReadReceiptAsync(ReadReceiptInfo readReceiptInfo, ITurnContext<IEventActivity> turnContext, CancellationToken cancellationToken) 
+
+```
+
+Following is an example of read receipts event request that a bot receives:
+
+```json
+{
+    "name": "application/vnd.microsoft.readReceipt",
+    "type": "event",
+    "timestamp": "2023-08-16T17:23:11.1366686Z",
+    "id": "f:b4783e72-9d7b-2ed9-ccef-ab446c873007",
+    "channelId": "msteams",
+    "serviceUrl": "https://canary.botapi.skype.com/amer/",
+    "from": {
+        "id": "29:1-8Iuh70W9pRqV8tQK8o2nVjxz33RRGDKLf4Bh7gKnrzN8s7e4vCyrFwjkPbTCX_Co8c4aXwWvq3RBLr-WkkVMw",
+        "aadObjectId": "ee6e242f-5885-4303-9c85-648d51a7609d"
+    },
+    "conversation": {
+        "conversationType": "personal",
+        "tenantId": "528dbe3f-15e0-4e37-84a1-00cc305847dd",
+        "id": "a:1xlimp68NSUxEqK0ap2rXuwC9ITauHgV2M4RaDPkeRhV8qMaFn-RyilMZ62YiVdqs8pp43yQaRKvv_U2S2gOS5nM-y_pOxVe4BW1qMGPtqD0Bv3pw-nJXF0zhDlZHMZ1Z"
+    },
+    "recipient": {
+        "id": "28:9901a8b6-4fef-428b-80b1-ddb59361adeb",
+        "name": "Quan Test Bot"
```

Comment:
nit: Can we change the name to "Test Bot"?

Reply:
Updated as Test Bot